### ISSUE TITLE
Test FMI with code export in the CI

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#11413]
After the excellent job by @mahge and @AnHeuermann on the CMAKE compilation, we should constantly test FMU export, both in Linux and in Windows. Maybe not on the entire testsuite, but on a selected set of models, so that if something breaks in the setup, we see that immediately. Otherwise the code export option will always be broken, one way or the other.

@adrpo, @mahge, @AnHeuermann, @arun3688 what do you think?

**Comment by casella** [comment]
Adding @dietmarw to the loop.

**Comment by casella** [comment]
We should of course also test FMI cpp export.

**Comment by AnHeuermann** [comment]
We have a very limited number of tests in https://github.com/OpenModelica/OpenModelica/tree/master/testsuite/special/FmuExportCrossCompile and https://github.com/OpenModelica/OpenModelica/tree/master/testsuite/openmodelica/fmi

Our PR's are not tested on Windows and we don't test the CMake export or recompilation on Windows.

**Comment by casella** [comment]
> Our PR's are not tested on Windows and we don't test the CMake export or recompilation on Windows.

Yep, I discussed this yesterday with @mahge. We need to buy a fast, modern machine for Windows testing, which should hopefully slash the Windows compile time substantially. Then, we should add tests of all possible FMI export options _on Windows_ (because that's where most people use FMI, like it or not) to the testsuite.

**Comment by casella** [comment]
I think we should expand the testing in the following directions:
- test CS and ME (we claim to support both)
- test on different OSs _including Windows_
- test with static and dynamic link libraries
- test FMUs with code export and CMAKE
- test generated FMUs both with OMSimulator _and_ PyFMI, so we can tell apart OMSimulator issues from FMI export issues, and also make sure that our FMI export is not relying on some holes of OMSimulators in the implementation of the FMI standard

**Comment by AnHeuermann** [comment]
https://github.com/OpenModelica/OpenModelica/pull/11435 will add a test for FMPy for four FMUs to all PRs.
If we want to test larger sections we should add this to our [OpenModelicaLibraryTesting](https://github.com/OpenModelica/OpenModelicaLibraryTesting) scripts.
I created a (not so minimal) Docker image with FMPy: https://github.com/AnHeuermann/fmpy-dockerimage

**Comment by casella** [comment]
@AnHeuermann where do I see that in the CI logs? And, which FMUs did you select?

**Comment by AnHeuermann** [comment]
E.g. in https://test.openmodelica.org/jenkins/blue/organizations/jenkins/OpenModelica/detail/master/4858/pipeline/1070
It's the `linux-FMPy` stage in pipeline `fmuchecker + FMPy + OMEdit`.

The FMUs are the ones we do the cross-compile tests on: https://github.com/OpenModelica/OpenModelica/tree/master/testsuite/special/FmuExportCrossCompile
  - FmuExportCrossCompile
  - RoomHeating_OM.RH
  - WaterTank.Control
  - WaterTank.TestSingleWaterTank
  - BouncingBall

**Comment by casella** [comment]
Should we add some more? For example:
- one test with a purely algebraic system
- one test with clocked variables
- one test with multibody and fixed state selection
- one test with multibody and variable state selection
